app/src/main/java/org/thoughtcrime/securesms/util/DateUtils.java

```diff
@@ -0,0 +1,267 @@
+    } else if (isWithin(timestamp, 6, TimeUnit.DAYS)) {
+      return getFormattedDateTime(timestamp, "EEE", locale);
+    } else if (isWithin(timestamp, 365, TimeUnit.DAYS)) {
+      return getFormattedDateTime(timestamp, "MMM d", locale);
+    } else {
+      return getFormattedDateTime(timestamp, "MMM d, yyyy", locale);
+    }
+  }
+
+  public static String getExtendedRelativeTimeSpanString(final Context c, final Locale locale, final long timestamp) {
+    if (isWithin(timestamp, 1, TimeUnit.MINUTES)) {
+      return c.getString(R.string.DateUtils_just_now);
+    } else if (isWithin(timestamp, 1, TimeUnit.HOURS)) {
+      int mins = (int)TimeUnit.MINUTES.convert(System.currentTimeMillis() - timestamp, TimeUnit.MILLISECONDS);
+      return c.getResources().getString(R.string.DateUtils_minutes_ago, mins);
+    } else {
+      StringBuilder format = new StringBuilder();
+      if      (isWithin(timestamp,   6, TimeUnit.DAYS)) format.append("EEE ");
+      else if (isWithin(timestamp, 365, TimeUnit.DAYS)) format.append("MMM d, ");
+      else                                              format.append("MMM d, yyyy, ");
+
+      if (DateFormat.is24HourFormat(c)) format.append("HH:mm");
+      else                              format.append("hh:mm a");
+
+      return getFormattedDateTime(timestamp, format.toString(), locale);
+    }
+  }
+
+  public static String getSimpleRelativeTimeSpanString(final Context context, final Locale locale, final long timestamp) {
+    if (isWithin(timestamp, 1, TimeUnit.MINUTES)) {
+      return context.getString(R.string.DateUtils_just_now);
+    } else if (isWithin(timestamp, 1, TimeUnit.HOURS)) {
+      int mins = (int) TimeUnit.MINUTES.convert(System.currentTimeMillis() - timestamp, TimeUnit.MILLISECONDS);
+      return context.getResources().getString(R.string.DateUtils_minutes_ago, mins);
+    } else {
+      String format = DateFormat.is24HourFormat(context) ? "HH:mm" : "hh:mm a";
+      return getFormattedDateTime(timestamp, format, locale);
+    }
+  }
+
+  public static String getTimeString(final Context c, final Locale locale, final long timestamp) {
+    StringBuilder format = new StringBuilder();
+
+    if      (isSameDay(System.currentTimeMillis(), timestamp)) format.append("");
+    else if (isWithinAbs(timestamp,   6, TimeUnit.DAYS))       format.append("EEE ");
+    else if (isWithinAbs(timestamp, 364, TimeUnit.DAYS))       format.append("MMM d, ");
+    else                                                       format.append("MMM d, yyyy, ");
+
+    if (DateFormat.is24HourFormat(c)) format.append("HH:mm");
+    else                              format.append("hh:mm a");
+
+    return getFormattedDateTime(timestamp, format.toString(), locale);
+  }
+
+  public static String getDayPrecisionTimeSpanString(Context context, Locale locale, long timestamp) {
+    SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyyMMdd");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.